<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Taxonomic resolution using lcvplants</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Taxonomic resolution using lcvplants</h1>



<div id="background" class="section level1">
<h1>Background</h1>
<p>When comparing or merging any list of plant species, for instance for
compiling regional species lists, merging occurrence lists with trait
data or phylogenies, the taxonomic names must be matched to avoid
artificial inflation due to synonyms, data loss or erroneous matches due
to homonyms. The <code>lcvplants</code> package facilitates this process
by automatizing large-scale taxonomic harmonization of plant names by
fuzzy matching and synonymy resolution against the Leipzig Catalogue of
Vascular Plants as taxonomic backbone.</p>
<p>The Leipzig Catalogue of Vascular Plants (LCVP) is a novel global
taxonomic backbone, updating The Plant List, comprising more than
1,300,000 names and 350,000 accepted taxa names. We described the LCVP
in detail in the related scientific publication (Freiberg et al,
2020).</p>
</div>
<div id="running-lcvplants" class="section level1">
<h1>Running lcvplants</h1>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>You can install <code>lcvplants</code> from github using the devtools
package You may need to install devtools first, so if you do not have it
installed, run the following code.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span></code></pre></div>
<p>To use <code>lcvplants</code> you also need the data of the <a href="https://github.com/idiv-biodiversity/LCVP"><code>LCVP</code>
package</a>, which you can install in the same way.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;idiv-biodiversity/lcvplants&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;idiv-biodiversity/LCVP&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lcvplants)</span></code></pre></div>
</div>
<div id="taxonomic-resolution" class="section level2">
<h2>Taxonomic resolution</h2>
<p>The basic function in <code>lcvplants</code> is
<code>lcvp_search</code>. It allows users to input a list of species
name that will be matched to the names in the LCVP data. It returns a
data.frame indicating the full name matched — including the authority
name and removing possible orthographic errors —, the taxonomic status,
the accepted name, family and order.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lcvp_search</span>(<span class="st">&quot;Hibiscus vitifolius&quot;</span>)</span></code></pre></div>
<table>
<colgroup>
<col width="9%" />
<col width="4%" />
<col width="5%" />
<col width="7%" />
<col width="3%" />
<col width="13%" />
<col width="6%" />
<col width="4%" />
<col width="12%" />
<col width="11%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Search</th>
<th align="right">global.Id</th>
<th align="left">Input.Genus</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="right">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Hibiscus vitifolius</td>
<td align="right">604882</td>
<td align="left">Hibiscus</td>
<td align="left">vitifolius</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">L.</td>
<td align="left">accepted</td>
<td align="right">604882</td>
<td align="left">Hibiscus vitifolius L.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>The <code>lcvp_search</code> algorithm will first try to exactly
match the binomial names provided by the user. If no match is found, it
will try to find the closest name given the maximal transformation cost
defined in the argument <code>max_distance</code> (see ?base::agrep for
more details). When the argument <code>show_correct = TRUE</code> a
column is added to the final result indicating whether the binomial name
was exactly matched (TRUE), or if it is misspelled (FALSE).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lcvp_search</span>(<span class="st">&quot;Hibiscus vitifoliuse&quot;</span>, </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">max_distance =</span> <span class="fl">0.1</span>, </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">show_correct =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="9%" />
<col width="4%" />
<col width="5%" />
<col width="7%" />
<col width="3%" />
<col width="12%" />
<col width="6%" />
<col width="4%" />
<col width="11%" />
<col width="10%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="4%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Search</th>
<th align="right">global.Id</th>
<th align="left">Input.Genus</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="right">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
<th align="left">Correct</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Hibiscus vitifoliuse</td>
<td align="right">604882</td>
<td align="left">Hibiscus</td>
<td align="left">vitifolius</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">L.</td>
<td align="left">accepted</td>
<td align="right">604882</td>
<td align="left">Hibiscus vitifolius L.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<p>If more than one name is fuzzy matched (more than one closest match
found), only the accepted or the first name will be returned. The
function <code>lcvp_fuzzy_search</code> can be used to return all the
closest results when argument <code>keep_closest = TRUE</code>, or all
names within the <code>max_distance</code> defined when
<code>keep_closest = FALSE</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lcvp_fuzzy_search</span>(<span class="st">&quot;Hibiscus vitifoliuse&quot;</span>, </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">max_distance =</span> <span class="fl">0.1</span>, </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">keep_closest =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table>
<colgroup>
<col width="5%" />
<col width="6%" />
<col width="8%" />
<col width="4%" />
<col width="13%" />
<col width="7%" />
<col width="4%" />
<col width="12%" />
<col width="11%" />
<col width="5%" />
<col width="4%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">global.Id</th>
<th align="left">Input.Genus</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="right">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
<th align="right">Name.Distance</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">604882</td>
<td align="left">Hibiscus</td>
<td align="left">vitifolius</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">L.</td>
<td align="left">accepted</td>
<td align="right">604882</td>
<td align="left">Hibiscus vitifolius L.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="right">604883</td>
<td align="left">Hibiscus</td>
<td align="left">vitifolius</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Mill.</td>
<td align="left">synonym</td>
<td align="right">603660</td>
<td align="left">Hibiscus cannabinus L.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p>Both <code>lcvp_search</code> and <code>lcvp_fuzzy_search</code>
allows multiple species search. If you need to run for many species
consider setting the argument <code>progress_bar = TRUE</code> to track
the execution progress.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>splist <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Hibiscus abelmoschus var. betulifolius Mast.&quot;</span>,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Hibiscus abutiloides Willd.&quot;</span>,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Hibiscus aculeatus&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Hibiscus acuminatus&quot;</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Hibiscus furcatuis&quot;</span> </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">lcvp_search</span>(splist, <span class="at">max_distance =</span> <span class="fl">0.2</span>, <span class="at">show_correct =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>x</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="17%" />
<col width="3%" />
<col width="4%" />
<col width="6%" />
<col width="3%" />
<col width="10%" />
<col width="5%" />
<col width="3%" />
<col width="9%" />
<col width="15%" />
<col width="3%" />
<col width="3%" />
<col width="4%" />
<col width="3%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Search</th>
<th align="right">global.Id</th>
<th align="left">Input.Genus</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="right">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
<th align="left">Correct</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Hibiscus abelmoschus var. betulifolius Mast.</td>
<td align="right">603424</td>
<td align="left">Hibiscus</td>
<td align="left">abelmoschus</td>
<td align="left">var.</td>
<td align="left">betulifolius</td>
<td align="left">Mast.</td>
<td align="left">synonym</td>
<td align="right">518</td>
<td align="left">Abelmoschus moschatus Medik.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Hibiscus abutiloides Willd.</td>
<td align="right">603426</td>
<td align="left">Hibiscus</td>
<td align="left">abutiloides</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Willd.</td>
<td align="left">synonym</td>
<td align="right">1211289</td>
<td align="left">Talipariti tiliaceum (L.) Fryxell</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Hibiscus aculeatus</td>
<td align="right">603439</td>
<td align="left">Hibiscus</td>
<td align="left">aculeatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Walter</td>
<td align="left">accepted</td>
<td align="right">603439</td>
<td align="left">Hibiscus aculeatus Walter</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Hibiscus acuminatus</td>
<td align="right">603440</td>
<td align="left">Hibiscus</td>
<td align="left">acuminatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Cav.</td>
<td align="left">synonym</td>
<td align="right">683056</td>
<td align="left">Kosteletzkya acuminata (Cav.) Britten</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Hibiscus furcatuis</td>
<td align="right">603937</td>
<td align="left">Hibiscus</td>
<td align="left">furcatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Roxb.</td>
<td align="left">accepted</td>
<td align="right">603937</td>
<td align="left">Hibiscus furcatus Roxb.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<p>The function <code>lcvp_summary</code> gives a report on the
searching results using the <code>lcvp_search</code> function.
Indicating the number of species searched, how many were matched. Among
the matched names, it indicates how many were exactly or fuzzy matched.
Then it checks how many author and infracategory names were exactly
matched. Note that if authors or infracategory is not provided, it will
be considered a no match.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lcvp_summary</span>(x)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Species searched: 5</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Species matched: 5 (100%)</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Species exactly matched: 4 (80%)</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Species fuzzy matched: 1 (20%)</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Authors exactly matched: 2 (40%)</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Infracategories exactly matched: 5 (100%)</span></span></code></pre></div>
<p>Notice that the warning message indicated that more than one name was
matched for some species. You can easily access these species using the
following code.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>sps_mult <span class="ot">&lt;-</span> <span class="fu">attr</span>(x, <span class="st">&quot;matched_mult&quot;</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>sps_mult</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Hibiscus aculeatus&quot; &quot;Hibiscus furcatuis&quot;</span></span></code></pre></div>
<p>And you can quickly use the <code>lcvp_fuzzy_search</code> on
them.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lcvp_fuzzy_search</span>(sps_mult)</span></code></pre></div>
<table>
<colgroup>
<col width="4%" />
<col width="5%" />
<col width="7%" />
<col width="3%" />
<col width="12%" />
<col width="6%" />
<col width="5%" />
<col width="11%" />
<col width="16%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="4%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">global.Id</th>
<th align="left">Input.Genus</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="right">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
<th align="right">Name.Distance</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">603436</td>
<td align="left">Hibiscus</td>
<td align="left">aculeatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">F.Dietr.</td>
<td align="left">synonym</td>
<td align="right">604041</td>
<td align="left">Hibiscus heterophyllus Vent.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">603437</td>
<td align="left">Hibiscus</td>
<td align="left">aculeatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">G.Don</td>
<td align="left">synonym</td>
<td align="right">604734</td>
<td align="left">Hibiscus surattensis L.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">603438</td>
<td align="left">Hibiscus</td>
<td align="left">aculeatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Roxb.</td>
<td align="left">synonym</td>
<td align="right">603937</td>
<td align="left">Hibiscus furcatus Roxb.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">603439</td>
<td align="left">Hibiscus</td>
<td align="left">aculeatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Walter</td>
<td align="left">accepted</td>
<td align="right">603439</td>
<td align="left">Hibiscus aculeatus Walter</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">603934</td>
<td align="left">Hibiscus</td>
<td align="left">furcatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Craib</td>
<td align="left">unresolved</td>
<td align="right">604561</td>
<td align="left">Hibiscus rosa-sinensis L.</td>
<td align="left">—</td>
<td align="left">—</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="right">603935</td>
<td align="left">Hibiscus</td>
<td align="left">furcatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Harv.</td>
<td align="left">synonym</td>
<td align="right">603462</td>
<td align="left">Hibiscus altissimus Hornby</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="right">603936</td>
<td align="left">Hibiscus</td>
<td align="left">furcatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Mullend.</td>
<td align="left">synonym</td>
<td align="right">604338</td>
<td align="left">Hibiscus noldeae Baker.f.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="right">603937</td>
<td align="left">Hibiscus</td>
<td align="left">furcatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Roxb.</td>
<td align="left">accepted</td>
<td align="right">603937</td>
<td align="left">Hibiscus furcatus Roxb.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="right">603940</td>
<td align="left">Hibiscus</td>
<td align="left">furcatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Wall.</td>
<td align="left">synonym</td>
<td align="right">603937</td>
<td align="left">Hibiscus furcatus Roxb.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="right">603941</td>
<td align="left">Hibiscus</td>
<td align="left">furcatus</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Willd.</td>
<td align="left">synonym</td>
<td align="right">604573</td>
<td align="left">Hibiscus rostellatus Guill. &amp; Perr.</td>
<td align="left">Malvaceae</td>
<td align="left">Malvales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
<div id="search-all-vascular-plant-names-of-a-genus-family-order-or-author" class="section level2">
<h2>Search all vascular plant names of a Genus, Family, Order or
Author</h2>
<p>Users can search all plant taxa names listed in the “Leipzig
Catalogue of Vascular Plants” (LCVP) by order, family, genus or author,
using the <code>lcvp_group_search</code> function.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Search by Genus</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">lcvp_group_search</span>(<span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;Adansonia&quot;</span>), <span class="at">search_by =</span> <span class="st">&quot;Genus&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<table>
<colgroup>
<col width="4%" />
<col width="5%" />
<col width="7%" />
<col width="3%" />
<col width="11%" />
<col width="7%" />
<col width="3%" />
<col width="10%" />
<col width="17%" />
<col width="5%" />
<col width="5%" />
<col width="12%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">global.Id</th>
<th align="left">Input.Genus</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="right">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">Aa</td>
<td align="left">argyrolepis</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Rchb.f.</td>
<td align="left">accepted</td>
<td align="right">1</td>
<td align="left">Aa argyrolepis (Rchb.f.) Rchb.f.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="left">Aa</td>
<td align="left">aurantiaca</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">D.Trujillo</td>
<td align="left">accepted</td>
<td align="right">2</td>
<td align="left">Aa aurantiaca D.Trujillo</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Lankesteriana 2011.11.1 1-8;</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="left">Aa</td>
<td align="left">brevis</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">synonym</td>
<td align="right">819078</td>
<td align="left">Myrosmodes breve (Schltr.) Garay</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="left">Aa</td>
<td align="left">calceata</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Schltr.</td>
<td align="left">accepted</td>
<td align="right">4</td>
<td align="left">Aa calceata (Rchb.f.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="right">5</td>
<td align="left">Aa</td>
<td align="left">chiogena</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">synonym</td>
<td align="right">819080</td>
<td align="left">Myrosmodes chiogena (Schltr.) C.A.Vargas</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="left">Aa</td>
<td align="left">colombiana</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">accepted</td>
<td align="right">6</td>
<td align="left">Aa colombiana Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Ann Bot 2009.104.3 403-416;</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>Users can choose to keep only certain taxonomic stauts, this includes
“accepted”, “synonym”, “unresolved”, and “external”.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Search by Author and keep only accepted names</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">lcvp_group_search</span>(<span class="st">&quot;Schltr.&quot;</span>, <span class="at">search_by =</span> <span class="st">&quot;Author&quot;</span>, <span class="at">status =</span> <span class="st">&quot;accepted&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="1%" />
<col width="4%" />
<col width="5%" />
<col width="7%" />
<col width="3%" />
<col width="12%" />
<col width="8%" />
<col width="4%" />
<col width="11%" />
<col width="14%" />
<col width="5%" />
<col width="5%" />
<col width="12%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">global.Id</th>
<th align="left">Input.Genus</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="right">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">2</td>
<td align="right">4</td>
<td align="left">Aa</td>
<td align="left">calceata</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Schltr.</td>
<td align="left">accepted</td>
<td align="right">4</td>
<td align="left">Aa calceata (Rchb.f.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">6</td>
<td align="left">Aa</td>
<td align="left">colombiana</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">accepted</td>
<td align="right">6</td>
<td align="left">Aa colombiana Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Ann Bot 2009.104.3 403-416;</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="right">7</td>
<td align="left">Aa</td>
<td align="left">denticulata</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">accepted</td>
<td align="right">7</td>
<td align="left">Aa denticulata Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="right">8</td>
<td align="left">Aa</td>
<td align="left">erosa</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Schltr.</td>
<td align="left">accepted</td>
<td align="right">8</td>
<td align="left">Aa erosa (Rchb.f.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="right">9</td>
<td align="left">Aa</td>
<td align="left">fiebrigii</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Schltr.) Schltr.</td>
<td align="left">accepted</td>
<td align="right">9</td>
<td align="left">Aa fiebrigii (Schltr.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="right">14</td>
<td align="left">Aa</td>
<td align="left">hieronymi</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Cogn.) Schltr.</td>
<td align="left">accepted</td>
<td align="right">14</td>
<td align="left">Aa hieronymi (Cogn.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
</tr>
</tbody>
</table>
<div id="basic-output-information" class="section level3">
<h3>Basic output information</h3>
<p>The output from <code>lcvp_search</code>,
<code>lcvp_fuzzy_search</code> and <code>lcvp_group_search</code> is a
<code>data.frame</code> (or list of data.frames) with the following
columns:</p>
<ul>
<li><em>global.Id</em>: The fixed species id of the input taxon in the
Leipzig Catalogue of Vascular Plants (LCVP).</li>
<li><em>Input.Genus</em>: A character vector. The input genus of the
corresponding vascular plant species name listed in LCVP.</li>
<li><em>Input.Epitheton</em>: A character vector. The input epitheton of
the corresponding vascular plant species name listed in LCVP.</li>
<li><em>Rank</em>: A character vector. The taxonomic rank (“species”,
subspecies: “subsp.”, variety: “var.”, subvariety: “subvar.”, “forma”,
or subforma: “subf.”) of the corresponding vascular plant species name
listed in LCVP.</li>
<li><em>Input.Subspecies.Epitheton</em>: A character vector. If the
indicated rank is below species, the subspecies epitheton input of the
corresponding vascular plant species name listed in LCVP. If the rank is
“species”, the input is “nil”.</li>
<li><em>Input.Authors</em>: A character vector. The taxonomic authority
input of the corresponding vascular plant species name listed in
LCVP.</li>
<li><em>Status</em>: A character vector. description if a taxon is
classified as ‘valid’, ‘synonym’, ‘unresolved’, ‘external’ or ‘blanks’.
The ‘unresolved’ rank means that the status of the plant name could be
either valid or synonym, but the information available does not allow a
definitive decision. ‘External’ in an extra rank which lists names
outside the scope of this publication but useful to keep on this updated
list. ‘Blanks’ means that the respective name exists in bibliography but
it is neither clear where it came from valid, synonym or unresolved.
(see the main text Freiberg et al. for more details)</li>
<li><em>globalId.of.Output.Taxon</em>: The fixed species id of the
output taxon in LCVP.</li>
<li><em>Output.Taxon</em>: A character vector. The list of the accepted
plant taxa names according to the LCVP.</li>
<li><em>Family</em>: A character vector. The corresponding family name
of the Input.Taxon, staying empty if the Status is unresolved.</li>
<li><em>Order</em>: A character vector. The corresponding order name of
the Input.Taxon, staying empty if the Status is unresolved.</li>
<li><em>Literature</em>: A character vector. The bibliography used.</li>
<li><em>Comments</em>: A character vector. Further taxonomic
comments.</li>
</ul>
<p>If no match is found for one species it will return NA for the
columns in the LCVP table. But, if no match is found for all species the
function will return NULL and a warning message.</p>
</div>
</div>
<div id="matching-and-joining-two-lists-of-vascular-plant-names" class="section level2">
<h2>Matching and joining two lists of vascular plant names</h2>
<div id="match" class="section level3">
<h3>Match</h3>
<p>In many situations, researchers want to compare and match two lists
of species name coming from different sources (e.g. phylogenies, spatial
data, trait data, regional lists). The function <code>lcvp_match</code>
can be used for that. It matches and compares two name lists based on
the taxonomic resolution of plant taxa names listed in the LCVP.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate two lists of species name</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>splist1 <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">apply</span>(LCVP<span class="sc">::</span>tab_lcvp[<span class="dv">2</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>], <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>))</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>splist2 <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">apply</span>(LCVP<span class="sc">::</span>tab_lcvp[<span class="dv">11</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>], <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>))</span></code></pre></div>
<p>Ordered is based on the splist1, and followed by non-matched names in
splist2.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Match both lists</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">lcvp_match</span>(splist1, splist2)</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<table>
<colgroup>
<col width="1%" />
<col width="5%" />
<col width="5%" />
<col width="3%" />
<col width="4%" />
<col width="5%" />
<col width="2%" />
<col width="9%" />
<col width="6%" />
<col width="3%" />
<col width="8%" />
<col width="10%" />
<col width="4%" />
<col width="4%" />
<col width="9%" />
<col width="3%" />
<col width="6%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Species.List.1</th>
<th align="left">Species.List.2</th>
<th align="left">global.Id</th>
<th align="left">Input.Genus</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="left">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
<th align="left">Match.Position.2to1</th>
<th align="right">Duplicated.Output.Position</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">6</td>
<td align="left">Aa colombiana</td>
<td align="left">Aa colombiana</td>
<td align="left">6</td>
<td align="left">Aa</td>
<td align="left">colombiana</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">accepted</td>
<td align="left">6</td>
<td align="left">Aa colombiana Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Ann Bot 2009.104.3 403-416;</td>
<td align="left"></td>
<td align="left">2</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">Aa denticulata</td>
<td align="left">Aa denticulata</td>
<td align="left">7</td>
<td align="left">Aa</td>
<td align="left">denticulata</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">accepted</td>
<td align="left">7</td>
<td align="left">Aa denticulata Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">5</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">Aa aurantiaca</td>
<td align="left">NA</td>
<td align="left">2</td>
<td align="left">Aa</td>
<td align="left">aurantiaca</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">D.Trujillo</td>
<td align="left">accepted</td>
<td align="left">2</td>
<td align="left">Aa aurantiaca D.Trujillo</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Lankesteriana 2011.11.1 1-8;</td>
<td align="left"></td>
<td align="left">NA</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">Aa figueroi</td>
<td align="left">Aa figueroi</td>
<td align="left">10</td>
<td align="left">Aa</td>
<td align="left">figueroi</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Szlach. &amp; S.Nowak</td>
<td align="left">accepted</td>
<td align="left">10</td>
<td align="left">Aa figueroi Szlach. &amp; S.Nowak</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">7</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">Aa calceata</td>
<td align="left">Aa calceata</td>
<td align="left">4</td>
<td align="left">Aa</td>
<td align="left">calceata</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Schltr.</td>
<td align="left">accepted</td>
<td align="left">4</td>
<td align="left">Aa calceata (Rchb.f.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="left">Aa fiebrigii</td>
<td align="left">Aa fiebrigii</td>
<td align="left">9</td>
<td align="left">Aa</td>
<td align="left">fiebrigii</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Schltr.) Schltr.</td>
<td align="left">accepted</td>
<td align="left">9</td>
<td align="left">Aa fiebrigii (Schltr.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">4</td>
<td align="right">NA</td>
</tr>
</tbody>
</table>
<p>If <code>include_all = FALSE</code>, non-matched names in splist2 are
not included. And users can use the column
<code>Match.Position.2to1</code> to reorder splist2 to match
splist1.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Match both lists</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>matchLists <span class="ot">&lt;-</span> <span class="fu">lcvp_match</span>(splist1, splist2, <span class="at">include_all =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>splist2_reordered <span class="ot">&lt;-</span> splist2[matchLists<span class="sc">$</span>Match.Position<span class="fl">.2</span>to1]</span></code></pre></div>
</div>
<div id="join" class="section level3">
<h3>Join</h3>
<p>The <code>lcvp_join</code> function provides a quicker way to join
two tables based on the taxonomic resolution of plant taxa names listed
in the LCVP. It is inspired by the <code>dplyr::join</code> function.
The function add the columns from a first table to a second table based
on the list of species name in both tables. It first standardizes the
species names in both tables based on the “Leipzig Catalogue of Vascular
Plants” (LCVP) using the algorithm in <code>lcvp_search</code>. These
standardized names of both tables are then matched using the algorithm
in <code>lcvp_match</code>. The argument <code>type</code> indicates the
kind of join to be done. The option “full” join will keep all species,
“left” return all rows from the first table and drops the non-matches
from the second, “right” do the same for the second table, and “inner”
keep only matched species.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data.frame1</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>splist1 <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">apply</span>(LCVP<span class="sc">::</span>tab_lcvp[<span class="dv">2</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>], <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>))</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>tbl1 <span class="ot">&lt;-</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="st">&quot;Species&quot;</span> <span class="ot">=</span> splist1, <span class="st">&quot;Trait1&quot;</span> <span class="ot">=</span> <span class="fu">runif</span>(<span class="fu">length</span>(splist1)))</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data.frame2</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>splist2 <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">apply</span>(LCVP<span class="sc">::</span>tab_lcvp[<span class="dv">11</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>], <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>))</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>tbl2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Species&quot;</span> <span class="ot">=</span> splist2,</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Trait2&quot;</span> <span class="ot">=</span> <span class="fu">runif</span>(<span class="fu">length</span>(splist2)),</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Trait3&quot;</span> <span class="ot">=</span> <span class="fu">runif</span>(<span class="fu">length</span>(splist2)),</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Trait4&quot;</span> <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="st">&quot;a&quot;</span>, <span class="st">&quot;b&quot;</span>), <span class="fu">length</span>(splist2), <span class="at">replace =</span> <span class="cn">TRUE</span>),</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Trait5&quot;</span> <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">FALSE</span>), <span class="fu">length</span>(splist2), <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Full join the two tables</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">lcvp_join</span>(tbl1, tbl2, <span class="fu">c</span>(<span class="st">&quot;Species&quot;</span>, <span class="st">&quot;Species&quot;</span>), <span class="at">type =</span> <span class="st">&quot;full&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="5%" />
<col width="5%" />
<col width="3%" />
<col width="5%" />
<col width="2%" />
<col width="9%" />
<col width="6%" />
<col width="3%" />
<col width="8%" />
<col width="11%" />
<col width="4%" />
<col width="4%" />
<col width="10%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="2%" />
<col width="2%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Species.List.1</th>
<th align="left">Species.List.2</th>
<th align="left">global.Id</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="left">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
<th align="right">Trait1</th>
<th align="right">Trait2</th>
<th align="right">Trait3</th>
<th align="left">Trait4</th>
<th align="left">Trait5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Aa colombiana</td>
<td align="left">Aa colombiana</td>
<td align="left">6</td>
<td align="left">colombiana</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">accepted</td>
<td align="left">6</td>
<td align="left">Aa colombiana Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Ann Bot 2009.104.3 403-416;</td>
<td align="left"></td>
<td align="right">0.9565583</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Aa brevis</td>
<td align="left">Aa brevis</td>
<td align="left">3</td>
<td align="left">brevis</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">synonym</td>
<td align="left">819078</td>
<td align="left">Myrosmodes breve (Schltr.) Garay</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0.7370195</td>
<td align="right">0.6833922</td>
<td align="right">0.8140077</td>
<td align="left">a</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Aa fiebrigii</td>
<td align="left">Aa fiebrigii</td>
<td align="left">9</td>
<td align="left">fiebrigii</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Schltr.) Schltr.</td>
<td align="left">accepted</td>
<td align="left">9</td>
<td align="left">Aa fiebrigii (Schltr.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0.8694374</td>
<td align="right">0.8911534</td>
<td align="right">0.3425689</td>
<td align="left">b</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Aa calceata</td>
<td align="left">Aa calceata</td>
<td align="left">4</td>
<td align="left">calceata</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Schltr.</td>
<td align="left">accepted</td>
<td align="left">4</td>
<td align="left">Aa calceata (Rchb.f.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0.7185541</td>
<td align="right">0.0320514</td>
<td align="right">0.3476672</td>
<td align="left">b</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Aa aurantiaca</td>
<td align="left">NA</td>
<td align="left">2</td>
<td align="left">aurantiaca</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">D.Trujillo</td>
<td align="left">accepted</td>
<td align="left">2</td>
<td align="left">Aa aurantiaca D.Trujillo</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Lankesteriana 2011.11.1 1-8;</td>
<td align="left"></td>
<td align="right">0.5003410</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Aa erosa</td>
<td align="left">Aa erosa</td>
<td align="left">8</td>
<td align="left">erosa</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Schltr.</td>
<td align="left">accepted</td>
<td align="left">8</td>
<td align="left">Aa erosa (Rchb.f.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0.1218760</td>
<td align="right">0.8616032</td>
<td align="right">0.0781148</td>
<td align="left">b</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<p>Because some names may turn out to be synonyms based on the LCVP
taxonomic resolution, users may opt to solve duplicated species names by
summarizing traits given provided functions for common classes of
variables (numeric, character, and logical). This can be done by turning
the option <code>solve_duplicated = TRUE</code>, and the algorithm will
combine duplicated output names (<code>Output.Taxon</code> column) based
on users-defined functions to summarize the information.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">lcvp_join</span>(tbl1, tbl2, <span class="fu">c</span>(<span class="st">&quot;Species&quot;</span>, <span class="st">&quot;Species&quot;</span>), <span class="at">type =</span> <span class="st">&quot;full&quot;</span>,</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">solve_duplicated =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="5%" />
<col width="5%" />
<col width="3%" />
<col width="5%" />
<col width="2%" />
<col width="9%" />
<col width="6%" />
<col width="3%" />
<col width="8%" />
<col width="11%" />
<col width="4%" />
<col width="4%" />
<col width="10%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="2%" />
<col width="2%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Species.List.1</th>
<th align="left">Species.List.2</th>
<th align="left">global.Id</th>
<th align="left">Input.Epitheton</th>
<th align="left">Rank</th>
<th align="left">Input.Subspecies.Epitheton</th>
<th align="left">Input.Authors</th>
<th align="left">Status</th>
<th align="left">globalId.of.Output.Taxon</th>
<th align="left">Output.Taxon</th>
<th align="left">Family</th>
<th align="left">Order</th>
<th align="left">Literature</th>
<th align="left">Comments</th>
<th align="right">Trait1</th>
<th align="right">Trait2</th>
<th align="right">Trait3</th>
<th align="left">Trait4</th>
<th align="left">Trait5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Aa colombiana</td>
<td align="left">Aa colombiana</td>
<td align="left">6</td>
<td align="left">colombiana</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">accepted</td>
<td align="left">6</td>
<td align="left">Aa colombiana Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Ann Bot 2009.104.3 403-416;</td>
<td align="left"></td>
<td align="right">0.9565583</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Aa brevis</td>
<td align="left">Aa brevis</td>
<td align="left">3</td>
<td align="left">brevis</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">Schltr.</td>
<td align="left">synonym</td>
<td align="left">819078</td>
<td align="left">Myrosmodes breve (Schltr.) Garay</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0.7370195</td>
<td align="right">0.6833922</td>
<td align="right">0.8140077</td>
<td align="left">a</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Aa fiebrigii</td>
<td align="left">Aa fiebrigii</td>
<td align="left">9</td>
<td align="left">fiebrigii</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Schltr.) Schltr.</td>
<td align="left">accepted</td>
<td align="left">9</td>
<td align="left">Aa fiebrigii (Schltr.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0.8694374</td>
<td align="right">0.8911534</td>
<td align="right">0.3425689</td>
<td align="left">b</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Aa calceata</td>
<td align="left">Aa calceata</td>
<td align="left">4</td>
<td align="left">calceata</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Schltr.</td>
<td align="left">accepted</td>
<td align="left">4</td>
<td align="left">Aa calceata (Rchb.f.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0.7185541</td>
<td align="right">0.0320514</td>
<td align="right">0.3476672</td>
<td align="left">b</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Aa aurantiaca</td>
<td align="left">NA</td>
<td align="left">2</td>
<td align="left">aurantiaca</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">D.Trujillo</td>
<td align="left">accepted</td>
<td align="left">2</td>
<td align="left">Aa aurantiaca D.Trujillo</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left">Lankesteriana 2011.11.1 1-8;</td>
<td align="left"></td>
<td align="right">0.5003410</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Aa erosa</td>
<td align="left">Aa erosa</td>
<td align="left">8</td>
<td align="left">erosa</td>
<td align="left">species</td>
<td align="left">nil</td>
<td align="left">(Rchb.f.) Schltr.</td>
<td align="left">accepted</td>
<td align="left">8</td>
<td align="left">Aa erosa (Rchb.f.) Schltr.</td>
<td align="left">Orchidaceae</td>
<td align="left">Asparagales</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0.1218760</td>
<td align="right">0.8616032</td>
<td align="right">0.0781148</td>
<td align="left">b</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<p>Because some users may want to solve duplicated names outside the
<code>lcvp_join</code> function, the algorithm is available as an
individual function names <code>lcvp_solve_dups</code>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data.frame with duplicated names and different traits</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>splist <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">apply</span>(LCVP<span class="sc">::</span>tab_lcvp[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>], <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>))</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>search <span class="ot">&lt;-</span> <span class="fu">lcvp_search</span>(splist)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>tbl <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">&quot;Species&quot;</span> <span class="ot">=</span> search<span class="sc">$</span>Output.Taxon,</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Trait1&quot;</span> <span class="ot">=</span> <span class="fu">runif</span>(<span class="fu">length</span>(splist)),</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Trait2&quot;</span> <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="st">&quot;a&quot;</span>, <span class="st">&quot;b&quot;</span>), <span class="fu">length</span>(splist), <span class="at">replace =</span> <span class="cn">TRUE</span>),</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Trait3&quot;</span> <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">FALSE</span>), <span class="fu">length</span>(splist), <span class="at">replace =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Solve with default parameters</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">lcvp_solve_dups</span>(tbl, <span class="dv">1</span>)</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<table>
<colgroup>
<col width="72%" />
<col width="11%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Species</th>
<th align="right">Trait1</th>
<th align="left">Trait2</th>
<th align="left">Trait3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Abacopteris macrophylla (Ching ex Y.X.Lin) S.E.Fawc.
&amp; A.R.Sm.</td>
<td align="right">0.1736717</td>
<td align="left">b</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Aa tenebrosa C.M.Martín &amp; G.J.Scrocchi</td>
<td align="right">0.5054807</td>
<td align="left">a</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Breynia macrantha (Hassk.) Chakrab. &amp;
N.P.Balakr.</td>
<td align="right">0.2560722</td>
<td align="left">b, a</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Aa macra Schltr.</td>
<td align="right">0.9905706</td>
<td align="left">a</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Cyclosorus cuspidatus (Blume) Copel.</td>
<td align="right">0.3427098</td>
<td align="left">a</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Cyclosorus asperus (C.Presl) B.K.Nayar &amp;
S.Kaur</td>
<td align="right">0.6993979</td>
<td align="left">b</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="parallel-programing-with-lcvplants" class="section level1">
<h1>Parallel programing with lcvplants</h1>
<p>Some users may want to use lcvplants functions to perform taxonomic
harmonization for many thousands of species and wish to use the entire
computational capacity available to speed up the processing time. Check
this <a href="https://idiv-biodiversity.github.io/lcvplants/articles/articles/Parallel-name-harmonization.html">article</a>,
where we show how to run lcvp_search or lcvp_fuzzy_search in parallel in
an efficient way to reduce computational time.</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p>Freiberg, M., Winter, M., Gentile, A. et al. LCVP, The Leipzig
catalogue of vascular plants, a new taxonomic reference list for all
known vascular plants. Sci Data 7, 416 (2020). <a href="https://doi.org/10.1038/s41597-020-00702-z" class="uri">https://doi.org/10.1038/s41597-020-00702-z</a></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
